homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -88,6 +91,34 @@ async def async_step_import(self, user_input=None):
         """Import a config entry."""
         return await self.async_step_user(user_input)
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered device."""
+        # Filter out items not having "keenetic' in their name
+        if (
+            "keenetic"
+            not in discovery_info.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "").lower()
+        ):
+            return self.async_abort(reason="not_keenetic_ndms2")
+
+        host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
+
+        if self._already_configured(host):
+            return self.async_abort(reason="already_configured")
+
+        await self.async_set_unique_id(host)
```

Comment:
Since you have a unique id (a bad one, but on non-the-less) you can just do this, and skip the _already_configured function.

```suggestion
        await self.async_set_unique_id(host)
        self._abort_if_unique_id_configured()        
```